evm/contracts/core/24-host/IBCCommitment.sol

```diff
@@ -1,202 +1,143 @@
+    bytes1 public constant CONNECTIONS = 0x02;
+    bytes1 public constant CHANNELS = 0x03;
+    bytes1 public constant PACKETS = 0x04;
+    bytes1 public constant PACKET_ACKS = 0x05;
+    bytes1 public constant NEXT_SEQ_SEND = 0x06;
+    bytes1 public constant NEXT_SEQ_RECV = 0x07;
+    bytes1 public constant NEXT_SEQ_ACK = 0x08;
+
+    function clientStatePath(
+        uint32 clientId
+    ) internal pure returns (bytes memory) {
+        return abi.encodePacked(CLIENT_STATE, clientId);
     }
 
     function consensusStatePath(
-        string memory clientId,
-        uint64 revisionNumber,
-        uint64 revisionHeight
+        uint32 clientId,
+        uint64 height
     ) internal pure returns (bytes memory) {
-        return abi.encodePacked(
-            "clients/",
-            clientId,
-            "/consensusStates/",
-            LibString.toString(revisionNumber),
-            "-",
-            LibString.toString(revisionHeight)
-        );
+        return abi.encodePacked(CONSENSUS_STATE, clientId, height);
     }
 
-    function connectionPath(string memory connectionId)
-        internal
-        pure
-        returns (bytes memory)
-    {
-        return abi.encodePacked("connections/", connectionId);
+    function connectionPath(
+        uint32 connectionId
+    ) internal pure returns (bytes memory) {
+        return abi.encodePacked(CONNECTIONS, connectionId);
     }
 
     function channelPath(
-        string memory portId,
-        string memory channelId
+        uint32 channelId
     ) internal pure returns (bytes memory) {
-        return abi.encodePacked(
-            "channelEnds/ports/", portId, "/channels/", channelId
-        );
+        return abi.encodePacked(CHANNELS, channelId);
     }
 
     function packetCommitmentPath(
-        string memory portId,
-        string memory channelId,
+        uint32 channelId,
         uint64 sequence
     ) internal pure returns (bytes memory) {
-        return abi.encodePacked(
-            "commitments/ports/",
-            portId,
-            "/channels/",
-            channelId,
-            "/sequences/",
-            LibString.toString(sequence)
-        );
-    }
-
-    function packetAcknowledgementCommitmentPath(
-        string memory portId,
-        string memory channelId,
-        uint64 sequence
+        return abi.encodePacked(PACKETS, channelId, sequence);
+    }
+
+    function batchPacketsCommitmentPath(
+        uint32 channelId,
+        bytes32 batchHash
     ) internal pure returns (bytes memory) {
-        return abi.encodePacked(
-            "acks/ports/",
-            portId,
-            "/channels/",
-            channelId,
-            "/sequences/",
-            LibString.toString(sequence)
-        );
-    }
-
-    function packetReceiptCommitmentPath(
-        string memory portId,
-        string memory channelId,
-        uint64 sequence
+        return abi.encodePacked(PACKETS, channelId, batchHash);
```

Comment:
does `abi.encodePacked` optimize down to the same bytecode as manually writing to memory?